src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -81,16 +84,80 @@ protected virtual bool ProcessIndividualPackageJTokens(string filePath, ISingleF
         {
             var name = packageJToken["name"].ToString();
             var version = packageJToken["version"].ToString();
-
+            var authorToken = packageJToken["author"];
+            
             if (!SemanticVersion.TryParse(version, out _))
             {
                 Logger.LogWarning($"Unable to parse version \"{version}\" for package \"{name}\" found at path \"{filePath}\". This may indicate an invalid npm package component and it will not be registered.");
                 return false;
             }
 
-            var detectedComponent = new DetectedComponent(new NpmComponent(name, version));
+            var npmComponent = new NpmComponent(name, version);
+            npmComponent.Author = GetAuthor(authorToken, name, filePath);
+
+            var detectedComponent = new DetectedComponent(npmComponent);
+            
             singleFileComponentRecorder.RegisterUsage(detectedComponent);
             return true;
         }
+
+        private NpmAuthor GetAuthor(JToken authorToken, string packageName, string filePath)
+        {
+            var authorString = authorToken?.ToString();
+            if (string.IsNullOrEmpty(authorString))
+            {
+                return null;
+            }
+
+            string authorName = null;
+            string authorEmail = null;
```

Comment:
nit: no need to initialise to `null`

```suggestion
            string authorName;
            string authorEmail;
```

Reply:
I had to initialize `authorEmail` to null, since all the code branches does not assign the value to it, and passing the uninitialized value to constructor causes compilation error.